gateway/api/serializers.py

```diff
@@ -188,3 +188,67 @@ class RuntimeJobSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = RuntimeJob
+
+
+class RunProgramSerializer(serializers.Serializer):
+    """
+    Program serializer for the /run end-point
+    """
+
+    title = serializers.CharField(max_length=255, allow_blank=False)
+    entrypoint = serializers.CharField(max_length=255)
+    artifact = serializers.FileField(allow_empty_file=False, use_url=False)
+    dependencies = serializers.CharField(allow_blank=False)
+    arguments = serializers.CharField(allow_blank=False)
+    config = serializers.CharField(allow_blank=False)
+
+    def to_representation(self, instance):
+        """
+        Transforms string `config` to JSON
+        """
+        representation = super().to_representation(instance)
+        representation["config"] = json.loads(representation["config"])
+        return representation
+
+    def retrieve_one_by_title(self, title, author):
+        """
+        This method returns a Program entry if it finds an entry searching by the title, if not None
+        """
+        return (
+            Program.objects.filter(title=title, author=author)
+            .order_by("-created")
+            .first()
+        )
+
+    def create(self, validated_data):
+        pass
+
+    def update(self, instance, validated_data):
+        pass
+
+
+class RunProgramModelSerializer(serializers.ModelSerializer):
+    """
+    Program model serializer for the /run end-point
+    """
+
+    arguments = serializers.CharField(read_only=True)
+    config = serializers.CharField(read_only=True)
```

Comment:
I wonder why these additional fields are necessary.  Are they used?

Reply:
They are needed because if you pass to the `ModelSerializer` the data from the request that data contains `arguments` and `config` and it fails if you don't specify that those keys are not going to be used by the serializer (you do that with `read_only=True`).

Reply:
It fails because the model doesn't have those attribute.